CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/utility/SaveResultsFileWriterNGTest.java

```diff
@@ -53,48 +82,152 @@ public SaveResultsFileWriterNGTest() {
+     * valid directory. Confirm file is created with expected contents and that no exception is thrown. confirm
+     * ConstellationLoggerHelper.exportPropertyBuilder called with appropriate SUCCESS status.
      */
     @Test
-    public void testGenerateFilenameSuffix2() {
+    public void testWriteRecordStore() {
         Plugin plugin = new ExampleClass();
-        String filename = SaveResultsFileWriter.generateFilename(plugin, "xml");
-        Assert.assertFalse(filename.endsWith("..xml"));
+        TabularRecordStore tabularRecordStore = new TabularRecordStore();
+        String key = "TEST1KEY";
+        String value = "TEST1VALUE";
+        tabularRecordStore.add();
+        tabularRecordStore.set(key, value);
+
+        // This test actually creates the output file in user.home directory, checksa its contents, and then removes it.
+        mockedDataAccessPreferenceKeys.when(DataAccessPreferenceKeys::getDataAccessResultsDir).thenReturn(new File(System.getProperty("user.home")));
+        mockedRecordStoreUtilities.when(() -> RecordStoreUtilities.toCsv(Mockito.any(), Mockito.any())).thenCallRealMethod();
+        mockedConstellationLoggerHelper.when(() -> ConstellationLoggerHelper.exportPropertyBuilder(any(), any(), any(), any())).thenAnswer((var invocation) -> { 
+            Object[] args = invocation.getArguments();
+            File passedFile = (File)args[2];
+            constellationLoggerHelperStatus = (String)args[3];
+
+            // Ensure the file was created
+            fileCreated = passedFile.exists();
+            if (fileCreated) {
+                // Read file and confirm it contains 3 rows
+                int rows = 0;
+                Scanner reader = new Scanner(passedFile);
+                while (reader.hasNextLine()) {
+                    rows++;
+                    String data = reader.nextLine();
+                    if (rows == 1) dataValid = dataValid & (data.equals(key));
+                    if (rows == 2) dataValid = dataValid & (data.equals(value));
+                    if (rows >= 3) dataValid = false;
+                }
+                if (rows == 0) dataValid = false;
+                passedFile.delete();
+            }
+
+            // Return properties (which are ignored)
+            Properties properties = new Properties();
+            return properties;
+        });
+
+        try {
+            SaveResultsFileWriter.writeRecordStore(plugin, tabularRecordStore);
+            mockedRecordStoreUtilities.verify(times(1), () -> RecordStoreUtilities.toCsv(Mockito.any(), Mockito.any()));
+
+            Assert.assertTrue(fileCreated, "Record store file was created.");
+            Assert.assertEquals(constellationLoggerHelperStatus, ConstellationLoggerHelper.SUCCESS,
+                    "ConstellationLoggerHelper passed status = SUCCESS.");
+            Assert.assertTrue(dataValid, "Record contents matches expected.");
+            Assert.assertTrue(true, "Exception is not expected.");
```

Comment:
can also probably get away without needing this assert

Reply:
agreed.